### PR TITLE
#99 Warn users when they try to download HTTP files

### PR DESCRIPTION
**Pull request by JackGilmore**
## Description

Solves https://github.com/OpenDataScotland/the_od_bods/issues/99

## Motivation and Context

Trying to download files from HTTP only sites currently causes mixed content errors which makes downloading impossible without copying and pasting the download link into the browser address bar. Mixed content errors usually fail silently as the error only shows up in the browser's dev tools.

To mitigate this, I've added a listener where if a dataset resource link is clicked on and it is HTTP only, it will display a modal advising the user to copy and paste the URL into their browser address bar to download. It also provides a link to the original dataset page.

## How Has This Been Tested?

Tested trying to download a HTTP only resource e.g. [Angus Council Area Fact Sheet](http://localhost:4000/datasets/angus+council-angus+council+area+fact+sheet/). Modal appears as expected.

Tested trying to download a HTTPS resource. Still allows download as normal and modal does not appear.

## Screenshots (if appropriate):
image

## Types of changes

- [X] Bug fix (non-breaking change which fixes an issue)
- [ ] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to change)

## Checklist:

- [X] My code follows the code style of this project.
- [ ] My change requires a change to the documentation.
- [ ] I have updated the documentation accordingly.
- [X] I have read the **CONTRIBUTING** document.
- [ ] I have added tests to cover my changes.
- [ ] All new and existing tests passed.

### CODE REVIEW

File: scripts/src/templates/dataset-mixed-content-modal.js [hunk 1]
```diff
@@ -0,0 +1,21 @@
...
+    <div class="modal-content">
+      <div class="modal-header">
+        <button type="button" class="close" data-dismiss="modal" aria-label="Close"><span aria-hidden="true">&times;</span></button>
+        <h4 class="modal-title">Mixed content warning</h4>
+      </div> 
+      <div class="modal-body">
+        <p>This resource link is only served over HTTP. As such, we can not link to it as most modern browsers do not allow direct links from HTTPS secured sites to HTTP sites for security reasons.</p>
+        <p>To access the resource, please copy the following URL and paste it into the address bar of your browser:</p>
+        <pre>${resourceUrl}</pre>
+        <p>Alternatively, you can access the resource via the <a href="${originalDatasetUrl}" target="_blank">original dataset link</a>.</p>
```
Comment (KarenJewell) [inline scripts/src/templates/dataset-mixed-content-modal.js:18]: Original dataset url not quite working.
Screenshot 2023-02-22 at 23 07 34

File: scripts/src/templates/dataset-mixed-content-modal.js [hunk 2]
```diff
@@ -0,0 +1,21 @@
...
+  <div class="modal-dialog modal-lg" role="document">
+    <div class="modal-content">
+      <div class="modal-header">
+        <button type="button" class="close" data-dismiss="modal" aria-label="Close"><span aria-hidden="true">&times;</span></button>
+        <h4 class="modal-title">Mixed content warning</h4>
+      </div> 
+      <div class="modal-body">
+        <p>This resource link is only served over HTTP. As such, we can not link to it as most modern browsers do not allow direct links from HTTPS secured sites to HTTP sites for security reasons.</p>
+        <p>To access the resource, please copy the following URL and paste it into the address bar of your browser:</p>
+        <pre>${resourceUrl}</pre>
```
Comment (KarenJewell) [inline scripts/src/templates/dataset-mixed-content-modal.js]: wonder if we can add a "copy all" / "copy to clipboard" button here? It's too easy to make a mistake here and not copy the whole url.

**Review by KarenJewell — COMMENTED** [review]
1 bug and 1 feature, otherwise behaviour as expected 👍

File: scripts/src/templates/dataset-mixed-content-modal.js (hunk 1 above)
Comment (JackGilmore) [reply scripts/src/templates/dataset-mixed-content-modal.js:18]: Helps if I grab the URL correctly 🤦‍♂️ Fixed now!

File: scripts/src/templates/dataset-mixed-content-modal.js (hunk 2 above)
Comment (JackGilmore) [reply scripts/src/templates/dataset-mixed-content-modal.js]: mixed-content-clipboard
Good shout. Have implemented a Copy button.

Comment (KarenJewell) [reply scripts/src/templates/dataset-mixed-content-modal.js]: loving the screen recording - how handy!

**Review by KarenJewell — APPROVED** [review]

**Pull request merged**